### PR TITLE
MM-30986 Move global logger init until after server logger fully configured

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
Logging is configured in `(*app.Server).initLogging` where a newly created `mlog.Logger` is assigned as the server logger `(*app.Server).Log` and the global logger as well.  When assigning as the global logger via `mlog.InitGlobalLogger(s.Log)` a copy of the server logger is made instead of just assigning it.  I am not sure why it is copied - perhaps to avoid a race condition.  

However, later when the advanced logging is initialized, it was using the global logger APIs therefore working on the copy of the logger struct and only initializing for the global logger, not the server's copy.

This PR moves the `initGlobalLogger` call to the end of `initLogging` so a copy of a fully configured logger is made for global logging.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30986

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by reflog — APPROVED** [review]

**Review by isacikgoz — APPROVED** [review]
LGTM 👍

File: app/server.go [hunk 1]
```diff
@@ -598,18 +598,20 @@ func (s *Server) initLogging() error {
 		s.Log = mlog.NewLogger(utils.MloggerConfigFromLoggerConfig(&s.Config().LogSettings, utils.GetLogFileLocation))
 	}
 
+	// Use this app logger as the global logger (eventually remove all instances of global logging).
+	// This is deferred because a copy is made of the logger and it must be fully configured before
+	// the copy is made.
+	defer mlog.InitGlobalLogger(s.Log)
+
+	// Redirect default golang logger to this logger.
```
Comment (isacikgoz) [inline app/server.go:606]: nit:
```suggestion
	// Redirect default Go logger to this logger.
```

**Comment by wiggin77** [comment]
Hi @lindalumitchell .  Not sure if your test wants to QA this.  It would involve setting up a server instance with Advanced Logging enabled.  The change here just means a few more entries would end up in the log file when using Advanced Logging.

**Comment by lindalumitchell** [comment]
@wiggin77 thanks, I think this one is fine without QA on the PR.

**Pull request merged**